[Qtwebengine] add security patch

See here 
https://www.qt.io/blog/security-advisory-recently-reported-chromium-impacts-qtwebengine

Does this patch will sync on upstream? I try add patch wo my locally VCPKG, but I got a error like this:
![image](https://user-images.githubusercontent.com/94091114/162107407-d586b28c-8684-422b-ac4c-ba0617acd56c.png)

The error log:
```
ninja -t msvc -e environment.x64 -- "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.31.31103\bin\HostX64\x64/cl.exe" /nologo /showIncludes -DIS_TRACING_MOJOM_IMPL -DUSE_AURA=1 -DTOOLKIT_QT -D_HAS_EXCEPTIONS=0 -D__STD_C -D_CRT_RAND_S -D_CRT_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_DEPRECATE -D_ATL_NO_OPENGL -D_WINDOWS -DCERT_CHAIN_PARA_HAS_EXTRA_FIELDS -DPSAPI_VERSION=2 -DWIN32 -D_SECURE_ATL -DWINAPI_FAMILY=WINAPI_FAMILY_DESKTOP_APP -DWIN32_LEAN_AND_MEAN -DNOMINMAX -D_UNICODE -DUNICODE -DNTDDI_VERSION=NTDDI_WIN10_VB -D_WIN32_WINNT=0x0A00 -DWINVER=0x0A00 -D_DEBUG -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DU_USING_ICU_NAMESPACE=0 -DU_ENABLE_DYLOAD=0 -DUSE_CHROMIUM_ICU=1 -DU_ENABLE_TRACING=1 -DU_ENABLE_RESOURCE_TRACING=0 -DU_STATIC_IMPLEMENTATION -DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_FILE -DUCHAR_TYPE=wchar_t -Igen -I../../../../../src/bae72aec12-bb8d63e444.clean/src/3rdparty/chromium -I../../../../../src/bae72aec12-bb8d63e444.clean/src/3rdparty/chromium/third_party/perfetto/include -Igen/third_party/perfetto/build_config -Igen/third_party/perfetto -I../../../../../src/bae72aec12-bb8d63e444.clean/src/3rdparty/chromium/third_party/abseil-cpp -I../../../../../src/bae72aec12-bb8d63e444.clean/src/3rdparty/chromium/third_party/boringssl/src/include -I../../../../../src/bae72aec12-bb8d63e444.clean/src/3rdparty/chromium/third_party/protobuf/src -Igen/protoc_out -I../../../../../src/bae72aec12-bb8d63e444.clean/src/3rdparty/chromium/third_party/ced/src -I../../../../../src/bae72aec12-bb8d63e444.clean/src/3rdparty/chromium/third_party/icu/source/common -I../../../../../src/bae72aec12-bb8d63e444.clean/src/3rdparty/chromium/third_party/icu/source/i18n /Gy /FS /bigobj /utf-8 /Zc:sizedDealloc- /wd4117 /D__DATE__= /D__TIME__= /D__TIMESTAMP__= /W4 /wd4091 /wd4127 /wd4251 /wd4275 /wd4312 /wd4324 /wd4351 /wd4355 /wd4503 /wd4589 /wd4611 /wd4100 /wd4121 /wd4244 /wd4505 /wd4510 /wd4512 /wd4610 /wd4838 /wd4995 /wd4996 /wd4456 /wd4457 /wd4458 /wd4459 /wd4200 /wd4201 /wd4204 /wd4221 /wd4245 /wd4267 /wd4305 /wd4389 /wd4702 /wd4701 /wd4703 /wd4661 /wd4706 /wd4715 /wd4003 /wd4068 /wd5051 /Od /Ob0 /GF /Zi /MDd /TP /wd4577 /GR- /c gen/services/tracing/public/mojom/perfetto_service.mojom.cc /Foobj/services/tracing/public/mojom/mojom/perfetto_service.mojom.obj /Fd"obj/services/tracing/public/mojom/mojom_cc.pdb"
gen/services/tracing/public/mojom/perfetto_service.mojom.cc(986): fatal error C1001: Internal compiler error.
(compiler file 'msc1.cpp', line 1691)
 To work around this problem, try simplifying or changing the program near the locations listed above.
If possible please provide a repro here: https://developercommunity.visualstudio.com 
Please choose the Technical Support command on the Visual C++ 
 Help menu, or open the Technical Support help file for more information
E:\Lily\0310\vcpkg\buildtrees\qtwebengine\src\bae72aec12-bb8d63e444.clean\src\3rdparty\chromium\base/hash/md5_constexpr_internal.h(317): note: while evaluating constexpr function 'base::internal::MD5CE::Hash32'
gen/services/tracing/public/mojom/perfetto_service.mojom.cc(985): note: while evaluating constexpr function 'base::MD5Hash32Constexpr'
INTERNAL COMPILER ERROR in 'C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.31.31103\bin\HostX64\x64\cl.exe'
    Please choose the Technical Support command on the Visual C++
    Help menu, or open the Technical Support help file for more information
```

--- a/ports/qtwebengine/0ce5e91.diff
+++ b/ports/qtwebengine/0ce5e91.diff
@@ -1,0 +1,94 @@
+From 0ce5e91bdfa2cd7cac247911b9e8c4404c114937 Mon Sep 17 00:00:00 2001
+From: Peter Varga <pvarga@inf.u-szeged.hu>
+Date: Thu, 21 Apr 2022 08:49:53 +0200
+Subject: [PATCH] Workaround MSVC2022 ICE in constexpr functions
+
+It happens around initialization of STL containers in a constexpr
+function. In this case, aggregate initialization of std::array with
+double braces seems to cause the crash.
+
+For some reason it doesn't seem to happen in 98-based. This workaround
+can be reverted after Microsoft fixes the issue:
+https://developercommunity.visualstudio.com/t/fatal-error-C1001:-Internal-compiler-err/1669485
+
+Change-Id: I6bc2c71d328691cc74bc53c6d62f3d5df519b81e
+Pick-to: 90-based
+Fixes: QTBUG-101917
+Reviewed-by: Allan Sandfeld Jensen <allan.jensen@qt.io>
+---
+
+diff --git a/chromium/base/hash/md5_constexpr_internal.h b/chromium/base/hash/md5_constexpr_internal.h
+index b705bc8..5c9c004 100644
+--- a/chromium/base/hash/md5_constexpr_internal.h
++++ b/chromium/base/hash/md5_constexpr_internal.h
+@@ -281,15 +281,63 @@
+     return IntermediateDataToMD5Digest(ProcessMessage(data, n));
+   }
+ 
+-  static constexpr uint64_t Hash64(const char* data, uint32_t n) {
+-    IntermediateData intermediate = ProcessMessage(data, n);
+-    return (static_cast<uint64_t>(SwapEndian(intermediate.a)) << 32) |
+-           static_cast<uint64_t>(SwapEndian(intermediate.b));
++  static constexpr uint64_t Hash64(const char* message, uint32_t n) {
++    const uint32_t m = GetPaddedMessageLength(n);
++    IntermediateData intermediate0 = kInitialIntermediateData;
++    for (uint32_t offset = 0; offset < m; offset += 64) {
++      RoundData data = {
++          GetPaddedMessageWord(message, n, m, offset),
++          GetPaddedMessageWord(message, n, m, offset + 4),
++          GetPaddedMessageWord(message, n, m, offset + 8),
++          GetPaddedMessageWord(message, n, m, offset + 12),
++          GetPaddedMessageWord(message, n, m, offset + 16),
++          GetPaddedMessageWord(message, n, m, offset + 20),
++          GetPaddedMessageWord(message, n, m, offset + 24),
++          GetPaddedMessageWord(message, n, m, offset + 28),
++          GetPaddedMessageWord(message, n, m, offset + 32),
++          GetPaddedMessageWord(message, n, m, offset + 36),
++          GetPaddedMessageWord(message, n, m, offset + 40),
++          GetPaddedMessageWord(message, n, m, offset + 44),
++          GetPaddedMessageWord(message, n, m, offset + 48),
++          GetPaddedMessageWord(message, n, m, offset + 52),
++          GetPaddedMessageWord(message, n, m, offset + 56),
++          GetPaddedMessageWord(message, n, m, offset + 60)};
++      IntermediateData intermediate1 = intermediate0;
++      for (uint32_t i = 0; i < 64; ++i)
++        intermediate1 = ApplyStep(i, data, intermediate1);
++      intermediate0 = Add(intermediate0, intermediate1);
++    }
++    return (static_cast<uint64_t>(SwapEndian(intermediate0.a)) << 32) |
++           static_cast<uint64_t>(SwapEndian(intermediate0.b));
+   }
+ 
+-  static constexpr uint32_t Hash32(const char* data, uint32_t n) {
+-    IntermediateData intermediate = ProcessMessage(data, n);
+-    return SwapEndian(intermediate.a);
++  static constexpr uint32_t Hash32(const char* message, uint32_t n) {
++    const uint32_t m = GetPaddedMessageLength(n);
++    IntermediateData intermediate0 = kInitialIntermediateData;
++    for (uint32_t offset = 0; offset < m; offset += 64) {
++      RoundData data = {
++          GetPaddedMessageWord(message, n, m, offset),
++          GetPaddedMessageWord(message, n, m, offset + 4),
++          GetPaddedMessageWord(message, n, m, offset + 8),
++          GetPaddedMessageWord(message, n, m, offset + 12),
++          GetPaddedMessageWord(message, n, m, offset + 16),
++          GetPaddedMessageWord(message, n, m, offset + 20),
++          GetPaddedMessageWord(message, n, m, offset + 24),
++          GetPaddedMessageWord(message, n, m, offset + 28),
++          GetPaddedMessageWord(message, n, m, offset + 32),
++          GetPaddedMessageWord(message, n, m, offset + 36),
++          GetPaddedMessageWord(message, n, m, offset + 40),
++          GetPaddedMessageWord(message, n, m, offset + 44),
++          GetPaddedMessageWord(message, n, m, offset + 48),
++          GetPaddedMessageWord(message, n, m, offset + 52),
++          GetPaddedMessageWord(message, n, m, offset + 56),
++          GetPaddedMessageWord(message, n, m, offset + 60)};
++      IntermediateData intermediate1 = intermediate0;
++      for (uint32_t i = 0; i < 64; ++i)
++        intermediate1 = ApplyStep(i, data, intermediate1);
++      intermediate0 = Add(intermediate0, intermediate1);
++    }
++    return SwapEndian(intermediate0.a);
+   }
+ };
+ 

--- a/ports/qtwebengine/portfile.cmake
+++ b/ports/qtwebengine/portfile.cmake
@@ -85,13 +85,14 @@ endif()
 string(LENGTH "${CURRENT_BUILDTREES_DIR}" buildtree_length)
 # We know that C:/buildrees/${PORT} is to long to build Release. Debug works however. Means 24 length is too much but 23 might work. 
 if(buildtree_length GREATER 22 AND VCPKG_TARGET_IS_WINDOWS)
-    message(WARNING "Buildtree path is probably too long. Consider passing --x-buildtrees-root=<shortpath> to vcpkg!\nTrying to use '${CURRENT_BUILDTREES_DIR}/../tmp'")
+    message(WARNING "Buildtree path '${CURRENT_BUILDTREES_DIR}' is too long.\nConsider passing --x-buildtrees-root=<shortpath> to vcpkg!\nTrying to use '${CURRENT_BUILDTREES_DIR}/../tmp'")
     set(CURRENT_BUILDTREES_DIR "${CURRENT_BUILDTREES_DIR}/../tmp") # activly avoid long path issues in CI. -> Means CI will not return logs
     cmake_path(NORMAL_PATH CURRENT_BUILDTREES_DIR)
     string(LENGTH "${CURRENT_BUILDTREES_DIR}" buildtree_length_new)
     if(buildtree_length_new GREATER 22)
          message(FATAL_ERROR "Buildtree path is too long. Build will fail! Pass --x-buildtrees-root=<shortpath> to vcpkg!")
     endif()
+    file(MAKE_DIRECTORY "${CURRENT_BUILDTREES_DIR}")
 endif()
 
 ### Download third_party modules

--- a/ports/qtwebengine/portfile.cmake
+++ b/ports/qtwebengine/portfile.cmake
@@ -82,15 +82,17 @@ if(WIN32) # WIN32 HOST probably has win_flex and win_bison!
     endif()
 endif()
 
-#set(CURRENT_BUILDTREES_DIR "${CURRENT_BUILDTREES_DIR}/../tmp") # avoid long path issues in CI. 
-#cmake_path(NORMAL_PATH CURRENT_BUILDTREES_DIR)
-#file(MAKE_DIRECTORY "${CURRENT_BUILDTREES_DIR}")
+set(CURRENT_BUILDTREES_DIR "${CURRENT_BUILDTREES_DIR}/../tmp") # avoid long path issues in CI. 
+cmake_path(NORMAL_PATH CURRENT_BUILDTREES_DIR)
+file(MAKE_DIRECTORY "${CURRENT_BUILDTREES_DIR}")
 
 ### Download third_party modules
 vcpkg_from_git(
     OUT_SOURCE_PATH SOURCE_PATH_WEBENGINE
     URL git://code.qt.io/qt/qtwebengine-chromium.git
     REF "${${PORT}_chromium_REF}"
+    PATCHES
+        0ce5e91.diff
 )
 
 ##### qt_install_submodule

--- a/ports/qtwebengine/portfile.cmake
+++ b/ports/qtwebengine/portfile.cmake
@@ -82,9 +82,17 @@ if(WIN32) # WIN32 HOST probably has win_flex and win_bison!
     endif()
 endif()
 
-set(CURRENT_BUILDTREES_DIR "${CURRENT_BUILDTREES_DIR}/../tmp") # avoid long path issues in CI. 
-cmake_path(NORMAL_PATH CURRENT_BUILDTREES_DIR)
-file(MAKE_DIRECTORY "${CURRENT_BUILDTREES_DIR}")
+string(LENGTH "${CURRENT_BUILDTREES_DIR}" buildtree_length)
+# We know that C:/buildrees/${PORT} is to long to build Release. Debug works however. Means 24 length is too much but 23 might work. 
+if(buildtree_length GREATER 22 AND VCPKG_TARGET_IS_WINDOWS)
+    message(WARNING "Buildtree path is probably too long. Consider passing --x-buildtrees-root=<shortpath> to vcpkg!\nTrying to use '${CURRENT_BUILDTREES_DIR}/../tmp'")
+    set(CURRENT_BUILDTREES_DIR "${CURRENT_BUILDTREES_DIR}/../tmp") # activly avoid long path issues in CI. -> Means CI will not return logs
+    cmake_path(NORMAL_PATH CURRENT_BUILDTREES_DIR)
+    string(LENGTH "${CURRENT_BUILDTREES_DIR}" buildtree_length_new)
+    if(buildtree_length_new GREATER 22)
+         message(FATAL_ERROR "Buildtree path is too long. Build will fail! Pass --x-buildtrees-root=<shortpath> to vcpkg!")
+    endif()
+endif()
 
 ### Download third_party modules
 vcpkg_from_git(

--- a/ports/qtwebengine/vcpkg.json
+++ b/ports/qtwebengine/vcpkg.json
@@ -2,6 +2,7 @@
   "$comment": "x86-windows is not within the upstream support matrix of Qt6",
   "name": "qtwebengine",
   "version": "6.3.0",
+  "port-version": 1,
   "description": "Qt WebEngine",
   "homepage": "https://www.qt.io/",
   "license": null,

--- a/scripts/ci.baseline.txt
+++ b/scripts/ci.baseline.txt
@@ -927,9 +927,6 @@ qpid-proton:x64-uwp=fail
 qpid-proton:x64-windows-static=fail
 qt5-base:arm64-windows=fail
 
-# qtwebengine:x64-windows has an ICE in VS2022
-qtwebengine:x64-windows=fail
-
 # Skip deprecated Qt module
 # (remove after 1 year or longer due to vcpkg upgrade not handling removed ports correctly)
 qt5-canvas3d:x64-linux=skip

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5998,7 +5998,7 @@
     },
     "qtwebengine": {
       "baseline": "6.3.0",
-      "port-version": 0
+      "port-version": 1
     },
     "qtwebsockets": {
       "baseline": "6.3.0",

--- a/versions/q-/qtwebengine.json
+++ b/versions/q-/qtwebengine.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "959e01f5131b57a9949781c73252bcc71142ea8a",
+      "git-tree": "6562d0a0294290afd20da039ff60602c1f3f4701",
       "version": "6.3.0",
       "port-version": 1
     },

--- a/versions/q-/qtwebengine.json
+++ b/versions/q-/qtwebengine.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "959e01f5131b57a9949781c73252bcc71142ea8a",
+      "version": "6.3.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "47cd0b321b5eda832afbb46e58159ca3adb65ce9",
       "version": "6.3.0",
       "port-version": 0

